File: groupAnagrams.java
import java.util.*;

class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> res = new HashMap<>();
        for(String s : strs) {
            char[] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String sorted = new String(charArray);
            res.putIfAbsent(sorted, new ArrayList<>());
            res.get(sorted).add(s);
        }
        return new ArrayList<>(res.values());
    }
}

File: GroupAnagrams.go
func groupAnagrams(strs []string) [][]string {
	resp := make(map[string][]string)
	var res [][]string
	for _, str := range strs {
		runes := []rune(str)

		sort.Slice(runes, func(i, j int) bool {
			return runes[i] < runes[j]
		})

		sortedStr := string(runes)
		resp[sortedStr] = append(resp[sortedStr], str)
	}

	for _, val := range resp {
		res = append(res, val)
	}

	return res
}